docs/rfcs/2025-02-13-super-peer-groups-mvp.md

```diff
@@ -0,0 +1,148 @@
+- Sub protocol for inviting new members via sharing super-peer link (same considerations as above).
+- We may need more robust processing for forwarded messages - more events / edge cases.
+
+### 3. Optimize postgres client
+
+- Connection pool.
+- Optimization of indexes (different from SQLite) may be required.
+- Many small queries may have to be reworked into large queries in some cases.
+
+### 4. Availability of past messages, skipped messages - conversation pagination via requests to super-peers
+
+- Protocol to paginate group conversation.
+- Based on shared msg id? item ts?
+- Response batches messages.
+- Rate limit? Probably not MVP.
+
+## Super-peer agreement protocol
+
+This protocol draft is for larger scope MVP that includes short links.
+
+The main idea is that public groups should have identity, that is defined via a permanent link to them, and that only owner(s) should be able to control group identity and link. This link is a short link either to SMP blob (see simplexmq rfcs on blob extensions), or to an XFTP file (requires indefinitely long storage). For clarity, to distinguish from per super-peer group links used for establishing connections with super-peers, it will be further called "short link".
+
+Short link points to a blob/file containing:
+
+- Super-peer group links,
+- Owners' public keys for verifying ownership transfer and other administrative actions,
+- Other group metadata as required by further clarifications to protocol.
+
+UX for creating a public group should be straightforward:
+
+1. Select operators, whose super-peers to use, or optionally custom configured super-peers.
+2. Create group, which generates short link and invites super-peers to it. Connection progress is shown for each super-peer.
+3. Confirm creation, once super-peers connected. Super-peer can fail to connect, at least one connected super-peer is required for confirmation.
+4. Share link for members to join.
+
+Super-peers are pre-configured in app for preset operators. User can also add self-hosted or other known super-peers to custom configuration. Super-peer should have a SimpleX address to receive group join requests.
+
+Protocol for creating public group (happy path):
+
+1. Group owner's client (further owner) creates group record locally.
+2. Owner sends contact requests to selected super-peers SimpleX addresses. These contact requests are essentially invitations to be super-peers in this group. ConnInfo sent in these contact requests (INV) contains group invitation details (XGrpInv with added fields, or new specific protocol message - TBC).
+3. Super-peers receive these requests. They generate new group links specifically for this group, to serve as a point of connection to them for new members.
+4. Super-peers accept requests, sending generated group links in confirmation (CONF) ConnInfos.
```

Comment:
And the created connection is used as part of the group already for connecting an owner to super-peer.